blogs/2022/03/28/increase-productivity-with-containers.md

```diff
@@ -0,0 +1,89 @@
+
+One of my favorite things that industry professionals like to ask aspiring developers is, “how many lines of code does the average developer write per day?” Most guess in the hundreds or thousands – they’re always shocked to hear that the actual average figure lies in the tens.
+
+Then what are developers doing with the rest of their time? Sure, some of the remaining time is spent on code design and googling “how to center a div in css”, but an inordinate amount is pure overhead – setting up projects, onboarding other developers, and troubleshooting issues that you can’t seem to repro on your own machine.
+
+The VS Code team has used this as the heart of our research over the years: if we can reduce time spent on overhead, then we can increase productivity time. We have a vision where developers don’t have to keep fighting the same battles over and over again. This means a consistent dev setup that endures never-ending churn of version upgrades, configuration changes, and hardware refreshes.
+
+But what’s the path look like to get us there? Let’s examine the journey to increase developer productivity that has led us from local development, to container-based development, to the cloud.
+
+## Starting out with local development
+
+Let’s begin where all developers start (and where a lot still are): local development. If you’ve uttered the phrase, “But it works on my computer!”, then chances are that you were in the local development phase. This means that everything for your development environment exists locally on your machine.
+
+![A shrug emoji saying "It works on my machine"](it-works-on-my-machine.png)
+
+Let’s paint a picture of what local development looks like in the real world. Bobby is so excited because he just landed his first software engineering job! On day one, he shows up ready to code. Instead, his boss hands him a stack of onboarding notes detailing all the steps that he needs to install to get the code up and running. He ends up spending his first full week doing nothing but running install commands, then being bounced from developer to developer to help troubleshoot his build issues, until he can finally run the project on his laptop.
+
+After a little while on the project, Bobby is delivering features like a champ. Then a security vulnerability test alerts his team that they need to update their .NET Core version. Bobby goes through the effort of upgrading the version on his Windows machine, updating the dependencies in their project, and testing any necessary code changes. He pushes his changes up and everything looks good! Or does it?
+
+As it turns out, a week later his teammate Susie gets back from being out of office and didn’t get the memo about upgrading. When she pulls the latest code, the build fails. After some back and forth figuring out who the heck broke the build while she was gone, she realizes that she still has the outdated version on her machine.
```

Comment:
Around this paragraph, it begins to feel like quite a bit of text. Is there any way to add images, diagrams, and/or shorten and combine some paragraphs?

Reply:
Updated this section to be shorter + rearranged memes